docker/stan-wasm-server/logic/compilation.py

```diff
@@ -0,0 +1,98 @@
+import tempfile
+import subprocess
+from hashlib import sha1
+from shutil import copy2, rmtree
+from pathlib import Path
+
+from .definitions import CompilationStatus
+from .locking import get_nonce, acquire_compilation_lock, release_compilation_lock
+
+
+COMPILATION_TIMEOUT = 60 * 5
+COMPILATION_OUTPUTS = ['main.js', 'main.wasm']
+
+
+def _compute_stan_program_hash(program_file: Path):
+    stan_program = program_file.read_text()
+    # TODO: replace stan_program with a canonical form
+    return sha1(stan_program.encode()).hexdigest()
+
+
+def make_canonical_model_dir(job_id: str, tinystan_dir: str):
+    stan_code_file_name = Path(job_id, 'main.stan')
```

Comment:
Agreed in general point, though in this specific instance I think functions in this file shouldn't be dealing with job directories at all--I rewrote so that generating the path to the source file is in `compilation_job_mgmt.py` and this function just takes the path to the source file.

(If we want to re-add the check that it's called "main.stan" we could do that here, but I think this is pretty controlled...)